lib/Target/AVR/AsmParser/AVRAsmParser.cpp

```diff
@@ -281,7 +235,7 @@ MatchAndEmitInstruction(SMLoc IDLoc, unsigned &Opcode,
 
 int AVRAsmParser::parseRegister() {
   const AsmToken &Tok = Parser.getTok();
-  int RegNum = -1;
+  int RegNum = 0;
```

Comment:
Pretty sure it's ok. `NoRegister` is zero and I looked at the generated function `MatchRegisterName(...)`. But good looking out there ;)